### PR TITLE
Add support for `STRUCT_EXTRACT`

### PR DESCRIPTION
**Pull request by acreux**
We use `EXTRACT_STRUCT` to extract a value from a ROW/STRUCT column.

DuckDB uses `struct_extract`/`   https://duckdb.org/docs/sql/functions/nested
Presto uses: `foo.bar` uses relation to read a row nested field

```
         # self.validate('SELECT "a"."x"."abc" FROM a', 'SELECT EXTRACT_STRUCT("x", 'abc') FROM a', identity=False, read="presto")
```
test does not pass

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -450,6 +450,10 @@ class DateStrToDate(Func):
     pass
 
 
+class ExtractStruct(Func):
```
Comment (tobymao) [inline sqlglot/expressions.py:453]: should we call it StructExtract to keep it inline with duckdb?

Comment (acreux) [reply sqlglot/expressions.py:453]: yes, let's do it

File: sqlglot/dialects.py [hunk 2]
```diff
@@ -267,6 +273,11 @@ def _ts_or_ds_to_date_str_sql(self, expression):
         this = self.sql(expression, 'this')
         return f"DATE_FORMAT(DATE_PARSE(SUBSTR({this}, 1, 10), '%Y-%m-%d'), '%Y-%m-%d')"
 
+    def __struct_extract_sql(self, expression):
```
Comment (tobymao) [inline sqlglot/dialects.py:276]: this should be placed outside so it can be reused

Comment (tobymao) [reply sqlglot/dialects.py:276]: we shoudln't have duplicate functions

**Review by tobymao — APPROVED** [review]

**Pull request merged**